pkg/rdt/rdt.go

```diff
@@ -95,7 +102,7 @@ func (r *control) GetClasses() []string {
 	return ret
 }
 
-func (r *control) SetProcessClass(class string, pids ...string) error {
+func (r control) setProcessClass(class string, pids ...string) error {
```

Comment:
Hmm... are you sure it is a good idea to change the signature of *all* these functions to have a *copied receiver* instead of a *pointer* ? including the ones which are essentially *setters/mutators* ?

Now those work because you only touch `map` members which are passed *by reference*. However, this might bite us in the arse later, if/when someone extends `control` with something else than a `map` or a `slice` and then carelessly copies the idea of using non-pointer receivers for setters.

I'd rather have them consistently as *pointer receivers* already now, even if a non-pointer would work as a side-effect of the underlying data type.

Reply:
OH my, you're very much right. Thanks for staying sharp!! A total brain fart from me. Now fixed.